gui-libs/gtk/gtk-4.12.3.ebuild

```diff
@@ -0,0 +1,228 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{10..12} )
+inherit gnome.org gnome2-utils meson optfeature python-any-r1 toolchain-funcs virtualx xdg
+
+DESCRIPTION="GTK is a multi-platform toolkit for creating graphical user interfaces"
+HOMEPAGE="https://www.gtk.org/ https://gitlab.gnome.org/GNOME/gtk/"
+
+LICENSE="LGPL-2+"
+SLOT="4"
+IUSE="aqua broadway cloudproviders colord cups examples ffmpeg gstreamer +introspection sysprof test vulkan wayland +X cpu_flags_x86_f16c"
+REQUIRED_USE="
+	|| ( aqua wayland X )
+	test? ( introspection )
+"
+
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~loong ~ppc ~ppc64 ~riscv ~sparc ~x86"
+
+COMMON_DEPEND="
+	>=dev-libs/glib-2.76.0:2
+	>=x11-libs/cairo-1.17.6[aqua?,glib,svg(+),X?]
+	>=x11-libs/pango-1.50.0[introspection?]
+	>=dev-libs/fribidi-1.0.6
+	>=media-libs/harfbuzz-2.6.0:=
+	>=x11-libs/gdk-pixbuf-2.30:2[introspection?]
+	media-libs/libpng:=
+	media-libs/tiff:=
+	media-libs/libjpeg-turbo:=
+	>=media-libs/libepoxy-1.4[egl,X(+)?]
+	>=media-libs/graphene-1.10.0[introspection?]
+	app-text/iso-codes
+	x11-misc/shared-mime-info
+
+	cloudproviders? ( net-libs/libcloudproviders )
+	colord? ( >=x11-misc/colord-0.1.9:0= )
+	cups? ( >=net-print/cups-2.0 )
+	ffmpeg? ( media-video/ffmpeg:= )
+	gstreamer? (
+		>=media-libs/gst-plugins-bad-1.12.3:1.0
+		>=media-libs/gst-plugins-base-1.12.3:1.0[opengl]
+	)
+	introspection? ( >=dev-libs/gobject-introspection-1.76:= )
+	vulkan? ( media-libs/vulkan-loader:= )
+	wayland? (
+		>=dev-libs/wayland-1.21.0
+		>=dev-libs/wayland-protocols-1.31
+		media-libs/mesa[wayland]
+		>=x11-libs/libxkbcommon-0.2
+	)
+	X? (
+		>=app-accessibility/at-spi2-core-2.46.0
+		media-libs/fontconfig
+		media-libs/mesa[X(+)]
+		x11-libs/libX11
+		>=x11-libs/libXi-1.8
+		x11-libs/libXext
+		>=x11-libs/libXrandr-1.5
+		x11-libs/libXcursor
+		x11-libs/libXfixes
+		x11-libs/libXdamage
+		x11-libs/libXinerama
+	)
+"
```

Comment:
I suggest sorting:
```
COMMON_DEPEND="
	app-text/iso-codes
	>=dev-libs/fribidi-1.0.6
	>=dev-libs/glib-2.76.0:2
	>=media-libs/graphene-1.10.0[introspection?]
	>=media-libs/harfbuzz-2.6.0:=
	>=media-libs/libepoxy-1.4[egl,X(+)?]
	media-libs/libjpeg-turbo:=
	media-libs/libpng:=
	media-libs/tiff:=
	>=x11-libs/cairo-1.17.6[aqua?,glib,svg(+),X?]
	>=x11-libs/gdk-pixbuf-2.30:2[introspection?]
	>=x11-libs/pango-1.50.0[introspection?]
	x11-misc/shared-mime-info

	cloudproviders? ( net-libs/libcloudproviders )
	colord? ( >=x11-misc/colord-0.1.9:0= )
	cups? ( >=net-print/cups-2.0 )
	ffmpeg? ( media-video/ffmpeg:= )
	gstreamer? (
		>=media-libs/gst-plugins-bad-1.12.3:1.0
		>=media-libs/gst-plugins-base-1.12.3:1.0[opengl]
	)
	introspection? ( >=dev-libs/gobject-introspection-1.76:= )
	vulkan? ( media-libs/vulkan-loader:= )
	wayland? (
		>=dev-libs/wayland-1.21.0
		>=dev-libs/wayland-protocols-1.31
		media-libs/mesa[wayland]
		>=x11-libs/libxkbcommon-0.2
	)
	X? (
		>=app-accessibility/at-spi2-core-2.46.0
		media-libs/fontconfig
		media-libs/mesa[X(+)]
		x11-libs/libX11
		x11-libs/libXcursor
		x11-libs/libXdamage
		x11-libs/libXext
		x11-libs/libXfixes
		>=x11-libs/libXi-1.8
		x11-libs/libXinerama
		>=x11-libs/libXrandr-1.5
	)
"
```

Reply:
We follow the same order of the build configuration, is better to follow up the changes 
What is the rationale behind this order?

Reply:
I usually follow the devmanual here: https://devmanual.gentoo.org/general-concepts/dependencies/index.html
However, I guess it's fine to sort after build configuration, especially since there can be exceptions.